service/src/main/kotlin/xyz/lebkuchenfm/external/storage/mongo/repositories/XSoundsMongoRepository.kt

```diff
@@ -39,6 +40,10 @@ class XSoundsMongoRepository(database: MongoDatabase) : XSoundsRepository {
         collection.insertOne(XSoundEntity(sound))
     }
 
+    override suspend fun replace(sound: XSound) {
+        collection.replaceOne(eq(XSoundEntity::name.name, sound.name), XSoundEntity(sound))
```

Comment:
WDYT about incrementing count when finding? 

```
override suspend fun findByName(name: String, updateCount: Boolean): XSound? {
    val filter = eq(XSoundEntity::name.name, name)

    return if (updateCount) {
        val increment = inc(XSoundEntity::timesPlayed.name, 1)
        val options = FindOneAndUpdateOptions().returnDocument(ReturnDocument.BEFORE)
        collection.findOneAndUpdate(filter, increment, options)?.toDomain()
    } else {
        collection.find(filter).firstOrNull()?.toDomain()
    }
}
```

Reply:
As per discussion on Discord - we should be doing `findOneAndUpdate` (because we live in multithreaded environment) but not in `findByName` only in different dedicated method (`incrementPlayCountByName`?)